src/main/java/org/opensearch/knn/index/codec/util/KNNCodecUtil.java

```diff
@@ -102,4 +111,20 @@ public static long getTotalLiveDocsCount(final BinaryDocValues binaryDocValues)
         }
         return totalLiveDocs;
     }
+
+    public static SpaceType getSpaceType(final ModelDao modelDao, final FieldInfo fieldInfo) {
+        final String spaceTypeString = fieldInfo.getAttribute(SPACE_TYPE);
+        if (StringUtils.isNotEmpty(spaceTypeString)) {
+            return SpaceType.getSpace(spaceTypeString);
+        }
+
+        final String modelId = fieldInfo.getAttribute(MODEL_ID);
+        if (StringUtils.isNotEmpty(modelId)) {
```

Comment:
I held off on adding this here for now just because ModelUtil gets the model metadata from the system index instead of the cluster state. I need to fix this, but will leave it for later.